lib/src/executor_layer/error.rs

```diff
@@ -0,0 +1,25 @@
+use super::super::storage_layer::error::StorageError;
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ExecutorError {
+    #[error("Message type mismatch")]
+    MessageTypeMismatch,
+
+    #[error("Table already exists")]
+    TableAlreadyExists,
+
+    #[error("Table not found")]
+    TableNotFound,
+
+    #[error("Underlying error: {0}")]
+    StorageError(StorageError),
+}
+
+impl<T: Into<StorageError>> From<T> for ExecutorError {
```

Comment:
There's a blanket `impl<T: From<U>> Into<T> for U` in `std`, so you don't need this
Generally, it's better to implement `From` where possible

Reply:
And, since you're working inside a single crate, an orphan rule doesn't apply, so there should be no problem in writing `impl From<stuff> for ExecutorError` in the module where `stuff` is defined

Reply:
> There's a blanket `impl<T: From<U>> Into<T> for U` in `std`, so you don't need this Generally, it's better to implement `From` where possible

I need this to transform bincode and marble errors into Executor errors easily

Reply:
Ohh, I see